src/Rules.Framework/Core/PropertiesDictionary.cs

```diff
@@ -0,0 +1,248 @@
+namespace Rules.Framework.Core
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Runtime.CompilerServices;
+
+    /// <summary>
+    /// A dictionary to hold the properties of a <see cref="IConditionNode{TConditionType}"/>.
+    /// </summary>
+    /// <seealso cref="IDictionary{TKey, TValue}"/>
+    public class PropertiesDictionary : IDictionary<string, object>, IEnumerable, IEnumerable<KeyValuePair<string, object>>
```

Comment:
I think this class can be sealed, right?

Reply:
Given it is a specific application of a dictionary, even though it is a public type, I think it is safe to seal it 👍